src/toil/batchSystems/kubernetes.py

```diff
@@ -395,25 +391,31 @@ def getUpdatedBatchJob(self, maxWait):
 
-            w = kubernetes.watch.Watch()
-            # TODO: block and wait for the jobs to update, until maxWait is hit
-            # For now just say we couldn't get anything
-            for pod in w.stream(self._genPods, timeout_seconds=maxwait):
-                # if pod status is terminated then check exit code    
-                if pod.status.container_statuses[0].state is 'terminated':
-                    if pod.status.container_statues[0].state.exit_code == 0:
-                        return (pod.metadata.name,\
-                                pod.status.container_statues[0].state.exit_code, \
-                                (pod.status.container_statuses[0].state.finished_at - \
-                                        pod.status.container_statues[0].state.started_at).total_seconds())
-                    # if job failed
-                    else:
-                        logger.warning(pod.status.container_status[0].state.reason,
-                                pod.status.container_statuses[0].state.exit_code)
-                        return None
-                else:
-                    continue
-            return None
+            w = kubernetes.watch.Watch()    
+
+            if self.enableWatching:
+                for j in self._ourJobObjects():
+                    logger.debug(j.spec.template.metadata.labels[u'job-name'], type(j.spec.template.metadata.labels[u'job-name']))
+                    for event in w.stream(self.coreApi.list_namespaced_pod, self.namespace, timeout_seconds=maxWait):
+                        job = event['object']
+                        if job.metadata.name.startswith(self.jobPrefix):
+                            logger.info("Event: %s %s %s" % (event['type'],event['object'].kind, event['object'].metadata.name))
+                            if job.status.phase == 'Failure' or job.status.phase == 'Succeeded':
+                                logger.info("FINISHED")
+                                logger.info(job.status.container_statuses[0].state.terminated.reason,
+                                    job.status.container_statuses[0].state.terminated.exit_code)
+                                jobID = int(job.metadata.owner_references[0].name[len(self.jobPrefix):])
+                                terminated = job.status.container_statuses[0].state.terminated
+                                runtime = (terminated.finished_at - terminated.started_at).total_seconds()
+                                result = (jobID, terminated.exit_code, runtime)
+                                # Deleting not Working 404 client.rest.ApiException
+                                self.batchApi.delete_namespaced_job(job.metadata.owner_references[0].name,
+                                                                    self.namespace,
+                                                                    propagation_policy='Foreground')
+                                                                    
```

Comment:
After you call `delete_namespaced_job` you need to call `_waitForJobDeath` to wait around for it to go away. Otherwise the job you just asked to delete can be sent back to you when you list jobs later, since it isn't deleted yet.